src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationFailureException;
+import mate.academy.exception.RegistrationFailureException;
+import mate.academy.lib.Inject;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationFailureException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        if (userFromDbOptional.isEmpty()) {
+            throw new AuthenticationFailureException("Authentication failed. "
+                    + "Please check your email and password and try again. "
+                    + "If you don't have an account, you can register a new user.");
+        }
+        User user = userFromDbOptional.get();
```

Comment:
consider using Optional#orElseThrow method to reduce code amount. 
```suggestion
        Optional<User> userFromDbOptional = userService.findByEmail(email);
        if (userFromDbOptional.isEmpty()) {
            throw new AuthenticationFailureException("Authentication failed. "
                    + "Please check your email and password and try again. "
                    + "If you don't have an account, you can register a new user.");
        }
        User user = userFromDbOptional.get();
```